src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -270,45 +281,114 @@ public override Task<bool> FixAsync(ICodeCleanUpScope scope, ICodeCleanUpExecuti
             }
         }
 
-        private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
+        private async Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, ICodeCleanUpExecutionContext context, CancellationToken cancellationToken)
         {
-            // TODO: this one will be implemented later
-            // https://github.com/dotnet/roslyn/issues/30165
             var hierarchy = hierarchyContent.Hierarchy;
             if (hierarchy == null)
             {
-                // solution
-                return Task.FromResult(false);
+                return await FixSolutionAsync(_workspace.CurrentSolution, context, cancellationToken).ConfigureAwait(true);
             }
 
             var itemId = hierarchyContent.ItemId;
             if (itemId == (uint)VSConstants.VSITEMID.Root)
             {
-                // Project
+                // Map the hierarchy to a ProjectId. For hierarchies mapping to multitargeted projects, we first try to
+                // get the project in the most recent active context, but fall back to the first target framework if no
+                // active context is available.
+                var hierarchyToProjectMap = _workspace.Services.GetRequiredService<IHierarchyItemToProjectIdMap>();
```

Comment:
have we considered apply clean up on all context for multitargeted projects? like the way @genlu  did for his sync namepsace to folder name stuff?

asking since context is only exposed through editor but not through solution explorer? or should we ask platform team to expose context in code clean up menu in solution explorer?

Reply:
I'm not sure which way would be considered most correct. The current approach aims to minimize conflicts that occur during the cleanup process.

Reply:
I agree. but asked since the code path from solution does handle all multi targetting projects.

Reply:
➡️ #34419